x-pack/legacy/plugins/actions/server/actions_config.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { tryCatch } from 'fp-ts/lib/Option';
+import { URL } from 'url';
+import { curry } from 'lodash';
+
+export enum WhitelistedHosts {
+  Any = '*',
+}
+
+export interface ActionsKibanaConfig {
+  enabled: boolean;
+  whitelistedHosts: string[];
+}
+
+export class NotWhitelistedError extends Error {
+  constructor(message: string) {
+    super(message); // 'Error' breaks prototype chain here
+    Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
+  }
+}
+
+export interface ActionsConfigurationUtilities {
+  isWhitelistedHostname: (hostname: string) => string | NotWhitelistedError;
+  isWhitelistedUri: (uri: string) => string | NotWhitelistedError;
+}
+
+const whitelistingErrorMessage = 'target url not in whitelist';
+const doesValueWhitelistAnyHostname = (whitelistedHostname: string): boolean =>
```

Comment:
Regarding `doesValueWhitelistAnyHostname` it's not about FP, it's about communicating intent.
Implementation is easier to communicate (they read the code) but intent is much harder (read the code, and the surrounding code, and how they fit together). I find it helpful to communicate intent in the name of functions rather than implementation.

Hope that helps